packages/core/src/fhir-deduplication/deduplicate-fhir.ts

```diff
@@ -206,18 +204,18 @@ export function deduplicateFhir(
     });
   }
 
-  deduplicatedBundle.entry = Object.entries(resourceArrays)
+  fhirBundle.entry = Object.entries(resourceArrays)
     .filter(([resourceType]) => resourceType !== "devices")
     .flatMap(([, resources]) => {
       const entriesArray = Array.isArray(resources) ? resources : [resources];
       return entriesArray
         .flatMap(v => v || [])
         .map(removeDuplicateReferences)
-        .map(entry => buildCompleteBundleEntry(entry, deduplicatedBundle.type));
+        .map(entry => buildCompleteBundleEntry(entry, fhirBundle.type));
     });
-  deduplicatedBundle.total = deduplicatedBundle.entry.length;
+  fhirBundle.total = fhirBundle.entry.length;
 
-  return deduplicatedBundle;
+  return fhirBundle;
```

Comment:
TODO: remove return statement since this is mutative + update tests and callsites